lib/rubocop/cop/style/empty_block_parameter.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for delimiters that are pipes or braces for empty
+      # block parameters. Delimiters for empty block parameters do not cause
+      # syntax errors, but it looks strange.
+      #
+      # @example
+      #   # bad
+      #   a do ||
+      #     do_something
+      #   end
+      #
+      #   # bad
+      #   a { || do_something }
+      #
+      #   # good
+      #   a do
+      #   end
+      #
+      #   # good
+      #   a { do_something }
+      class EmptyBlockParameter < Cop
+        MSG = 'Omit delimiters for the empty block parameters.'.freeze
+
+        def_node_matcher :empty_arguments?, <<-PATTERN
+          (block _ $(args) _)
+        PATTERN
+
+        def on_block(node)
+          empty_arguments?(node) do |args|
+            return unless args.loc.expression
```

Comment:
Can we put this as a meaningfully named method in `ArgsNode`? 🙂

Reply:
Probably. I'll try it.